modules/axtask/src/run_queue.rs

```diff
@@ -567,6 +610,21 @@ fn gc_entry() {
     }
 }
 
+#[cfg(feature = "smp")]
+pub(crate) unsafe fn finish_task_switch() {
```

Comment:
This can be divided into two, one is still `clear_prev_task_on_cpu`, and one is selecting run queue for the migrated task after `resched` in `migrate_current`, so that we don't need to check `MIGRATING_TASK` during normal task switching.

Reply:
> This can be divided into two, one is still `clear_prev_task_on_cpu`, and one is selecting run queue for the migrated task after `resched` in `migrate_current`, so that we don't need to check `MIGRATING_TASK` during normal task switching.

I understand your intension. 

BUT, the question is, when a task calls migrate_current, it's the next task that is reponsible to check whether `MIGRATING_TASK` is not `None`, we can not just put something like selecting run queue for the migrated task after `resched` in `migrate_current`

Reply:
Why? After `resched`, current task is the next task, just like `clear_prev_task_on_cpu` on `switch_to`.

Reply:
Yep, but it will run `switch_to` belonging to next task (or `task_entry` belongs to next task), which will never return to the `resched` in `migrate_current`.